### PR TITLE
Configure FE unit tests

### PR DESCRIPTION
**Pull request by martinkeimel**
# Background
This PR contains the FE unit tests configuration using Karma+Jasmine along with UTs for the `echoService` and the `echoViewModel`.

# Changes done
- Update readme file to indicate how to run the BE and FE projects
- Add karma and jasmine related dependencies
- Add `karma.config`
- Add specs for the `echoService` and the `echoViewModel`.

# Required reviewers
- @DanielAltamirano 
- @mravinale

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,9 +1,28 @@
 # P202 training
```
Comment (GustavoAlMena) [inline README.md:1]: Great job! Code review approved!

**Review by mravinale — COMMENTED** [review]
why instead of using jquery.promises use better https://github.com/kriskowal/q ?

**Comment by martinkeimel** [comment]
No particular reason @mravinale . Just to avoid adding a new dependency since jquery is already included in the project. Do you want me to add Q?

**Comment by mravinale** [comment]
@martinkeimel I think q interface is more standard, and cleaner, and I would try to avoid any jquery usage as much as possible even for ajax calls

**Review by GustavoAlMena — APPROVED** [review]
Code review approved!

**Pull request merged**